portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +38,83 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    // Let user request number of comments to fetch and display
+    // int reqComments = getReqComments(request);
+    // System.out.println(reqComments);
+    int reqComments = 5;
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment singleComment = new Comment(id, name, text, timestamp);
+      comments.add(singleComment);
+    }
+
+    // Ensure maxComments is not greater than the existing number of comments
+    if (comments.size() < reqComments) {
+     reqComments = comments.size();
+    }
+    
+    // Remove comments that do not fit within maxComments
+    for (int i = comments.size()-1; i > reqComments-1; i--) {
```

Comment:
nit: add spacing before and after the subtract operator
```
for (int i = comments.size() - 1; i > reqComments - 1; i--)
```

Reply:
Fix in following commit